pkg/rhos/ocpgwdeployer.go

```diff
@@ -313,10 +313,24 @@ func (d *ocpGatewayDeployer) Cleanup(status reporter.Interface) error {
 
 			status.Success("Successfully reconfigured the instance")
 		}
+	}
 
-		status.End()
+	status.Start("Cleanup failed machine sets")
+
+	// creating a machineset with a dummy index for cleanup
+	machineSet, err := d.initMachineSet("0")
+	if err != nil {
+		return status.Error(err, "failed to cleanup failed machine sets")
 	}
 
+	err = d.msDeployer.DeleteAll(machineSet, "-submariner-gw-")
```

Comment:
There is another issue as discussed on slack.

When the MachineSet is created by the corresponding node is not active and stays in provisioned state for some reason, subsequent operations to cleanup/prepare OSP cloud is giving the following error.

```
[sgaddam@localhost aws-ocp]$ oc get machines -A
NAMESPACE               NAME                                   PHASE         TYPE                               REGION      ZONE   AGE
openshift-machine-api   rhos-ovn-7jzrt-master-0                Running       PnTAE.CPU_4_Memory_16384_Disk_50   regionOne   nova   6h57m
openshift-machine-api   rhos-ovn-7jzrt-master-1                Running       PnTAE.CPU_4_Memory_16384_Disk_50   regionOne   nova   6h57m
openshift-machine-api   rhos-ovn-7jzrt-master-2                Running       PnTAE.CPU_4_Memory_16384_Disk_50   regionOne   nova   6h57m
openshift-machine-api   rhos-ovn-7jzrt-submariner-gw-0-c25vt   Provisioned   PnTAE.CPU_4_Memory_8192_Disk_50    regionOne   nova   53m
openshift-machine-api   rhos-ovn-7jzrt-worker-0-5pkkr          Running       PnTAE.CPU_4_Memory_16384_Disk_50   regionOne   nova   6h52m
openshift-machine-api   rhos-ovn-7jzrt-worker-0-j9jns          Running       PnTAE.CPU_4_Memory_16384_Disk_50   regionOne   nova   6h52m
```

```
[sgaddam@localhost aws-ocp]$ subctl cloud cleanup rhos --ocp-metadata rhos-ovn/metadata.json --kubeconfig rhos-ovn/auth/kubeconfig 
 ✓ Obtained infra ID "rhos-ovn-7jzrt" and project ID "" from OCP metadata file "rhos-ovn/metadata.json"
 ✓ Obtained region "" from environment variable OS_REGION_NAME
 ✓ Retrieving RHOS credentials from your RHOS configuration 
 ✓ Successfully removed the Submariner gateway configuration from the nodes
 ✗ Deleting the Submariner gateway security group 
 ✗ Failed to cleanup RHOS cloud: error deleting the Submariner gateway security group: error deleting the security group "rhos-ovn-7jzrt-submariner-gw-sg": Bad request with: [DELETE https://rhos-d.infra.prod.upshift.rdu2.redhat.com:13774/v2.1/os-security-groups/6657e458-6d02-4f45-807a-0e7279d44e32], error message: {"badRequest": {"code": 400, "message": "Security Group 6657e458-6d02-4f45-807a-0e7279d44e32 in use.\nNeutron server returns request_ids: ['req-75a38a46-dde4-45fb-8cdd-1dbd023ddf39']"}}

subctl version: devel
```